docs/terms/execution_engine.md

```diff
@@ -58,10 +64,57 @@ You will not need to directly access an Execution Engine.  Instead, you will con
 
 You will not need to directly instantiate an Execution Engine.  Instead, they are automatically created as a component in a Datasource.
 
+### Execution Engine init arguments
+
+- `name`
+- `caching`
+- `batch_spec_defaults` (is this needed?)
+- `batch_data_dict`
+- `validator`
+
+### Execution Engine Properties
+
+- `loaded_batch_data` (all "loaded" batches)
+- `active_batch_data_id`
+
+### Execution Engine Methods
+
+- `load_batch_data(batrch_id, batch_data)`
+- `resolve_metrics`: computes metric values
+- `get_compute_domain`: gets the compute domain for a particular type of intermediate metric.
+
 ### Configuration
 
 Execution Engines and their configurations are specified in the configurations of Datasources.  In the configuration for your Datasource, you will have an `execution_engine` key.  This is a dictionary which will have at the least a `class_name` key that indicates the Execution Engine that will be associated with the Datasource.  If you are using a custom Execution Engine from a Plugin, you will also need to include a `module_name` key.  
 
 If additional configuration is required by the Execution Engine, it will also be specified in the `execution_engine` configuration.  For example, the `SqlAlchemyExecutionEngine` will also expect the key `connection_string` as part of its configuration.
 
 For specifics on the required keys for a given Execution Engine, please see our [how-to guides for Connecting to Data](../guides/connecting_to_your_data/index.md).
+
+## More details
+
+### Design motivation: Validation flow
```

Comment:
Just checking - is this section incomplete?

Reply:
It is.  It looks like my last commit wasn't pushed; I've corrected that now.